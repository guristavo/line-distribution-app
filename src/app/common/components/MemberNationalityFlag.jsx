import React from 'react';
import PropTypes from 'prop-types';

// Import utilities
import constants from '../../../utils/constants';
import { bem } from '../../../utils';

const MemberNationalityFlag = ({ nationality, className }) => {
  const flag = constants.FLAGS_LIST_ICON[nationality] || 'unknown';
  const baseClasses = bem('member-nationality-flag', [flag]);

  return <span className={`${baseClasses} ${className}`} title={nationality} />;
};

MemberNationalityFlag.propTypes = {
  nationality: PropTypes.string,
  className: PropTypes.string,
};

MemberNationalityFlag.defaultProps = {
  nationality: 'UNKNOWN',
  className: '',
};

export default MemberNationalityFlag;
