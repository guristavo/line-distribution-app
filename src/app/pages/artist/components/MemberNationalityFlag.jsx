import React from 'react';
import PropTypes from 'prop-types';
import constants from '../../../../utils/constants';

const MemberNationaltyFlag = ({ nationality }) => {
  const flag = constants.FLAGS_LIST[nationality];

  let flagUrl = `${process.env.PUBLIC_URL}${
    constants.NATIONALITY_FLAG_URL
  }unknown.jpg`;

  if (flag) {
    flagUrl = `${process.env.PUBLIC_URL}${
      constants.NATIONALITY_FLAG_URL
    }${flag}.jpg`;
  }

  return <img className="card__flag" src={flagUrl} alt={nationality} />;
};

MemberNationaltyFlag.propTypes = {
  nationality: PropTypes.string,
};

MemberNationaltyFlag.defaultProps = {
  nationality: 'unknown',
};

export default MemberNationaltyFlag;
