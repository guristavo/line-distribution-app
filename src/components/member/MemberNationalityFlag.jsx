import React from 'react';
import PropTypes from 'prop-types';
import { FLAGS_LIST, NATIONALITY_FLAG_URL } from '../../constants';

const MemberNationaltyFlag = ({ nationality }) => {
  const flag = FLAGS_LIST[nationality];

  let flagUrl = `${process.env.PUBLIC_URL}${NATIONALITY_FLAG_URL}unknown.jpg`;

  if (flag) {
    flagUrl = `${process.env.PUBLIC_URL}${NATIONALITY_FLAG_URL}${flag}.jpg`;
  }

  return <img className="card__flag" src={flagUrl} alt={nationality} />;
};

MemberNationaltyFlag.propTypes = {
  nationality: PropTypes.string,
};

MemberNationaltyFlag.defaultProps = {
  nationality: 'unknown',
};

export default MemberNationaltyFlag;
