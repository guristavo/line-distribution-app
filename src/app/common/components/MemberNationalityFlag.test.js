import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import MemberNationalityFlag from './MemberNationalityFlag';

const FLAGS = [
  {
    nationality: 'KOREAN',
    flag: 'korea',
  },
  {
    nationality: 'CHINESE',
    flag: 'china',
  },
  {
    nationality: 'AMERICAN',
    flag: 'usa',
  },
  {
    nationality: 'BRAZILIAN',
    flag: 'brazil',
  },
  {
    nationality: 'UNKNOWN',
    flag: 'unknown',
  },
];

describe('common/MemberNationalityFlag', () => {
  afterEach(cleanup);

  it('renders', () => {
    const { asFragment } = render(
      <MemberNationalityFlag nationality="KOREAN" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  for (let i = 0; i < FLAGS.length; i++) {
    const { nationality, flag } = FLAGS[0];
    it(`renders the correct flag: ${nationality}`, () => {
      const { getByTitle } = render(
        <MemberNationalityFlag nationality={nationality} />
      );

      const element = getByTitle(nationality);
      const classList = element.classList.value;
      expect(element).toBeInTheDocument();
      expect(classList.includes('member-nationality-flag')).toBe(true);
      expect(classList.includes(`member-nationality-flag--${flag}`)).toBe(true);
    });
  }

  it('renders as unknown when nationality is not in the flags list constant', () => {
    const { getByTitle } = render(<MemberNationalityFlag nationality="BOLA" />);

    const element = getByTitle('BOLA');
    const classList = element.classList.value;
    expect(element).toBeInTheDocument();
    expect(classList.includes('member-nationality-flag')).toBe(true);
    expect(classList.includes('member-nationality-flag--unknown')).toBe(true);
  });
});
