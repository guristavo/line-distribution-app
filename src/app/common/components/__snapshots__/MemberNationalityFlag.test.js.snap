// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/MemberNationalityFlag renders 1`] = `
<DocumentFragment>
  <span
    class="member-nationality-flag member-nationality-flag--korea "
    title="KOREAN"
  />
</DocumentFragment>
`;
